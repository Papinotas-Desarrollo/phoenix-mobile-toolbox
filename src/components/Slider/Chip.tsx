// @flow
import PropTypes from 'prop-types';
import React, { FunctionComponent } from 'react';
import { StyleSheet, Text, TouchableOpacity } from 'react-native';
import colors from '../../styles/colors';

const styles = StyleSheet.create({
  containerChip: {
    padding: 10,
    marginHorizontal: 5,
  },
  selectedChip: {
    borderBottomColor: colors.papinotasOrange,
    borderBottomWidth: 2,
  },
  selectedTextChip: {
    color: colors.slider.active,
    fontWeight: '600',
  },
  textProps: {
    color: colors.slider.inactive,
    fontSize: 14,
  },
});

interface ChipProps {
  name: string;
  selected: boolean;
  onPressSelection: Function;
}

const Chip: FunctionComponent<ChipProps> = ({ id, name, selected, onPressSelection }) => (
  <TouchableOpacity
    onPress={() => onPressSelection(id || name)}
    style={[styles.containerChip, selected ? styles.selectedChip : null]}
    disabled={selected}
  >
    <Text style={[styles.textProps, selected ? styles.selectedTextChip : null]}>
      {name}
    </Text>
  </TouchableOpacity>
);

Chip.propTypes = {
  name: PropTypes.number.isRequired,
  selected: PropTypes.bool.isRequired,
  onPressSelection: PropTypes.func.isRequired,
};

export default Chip;
