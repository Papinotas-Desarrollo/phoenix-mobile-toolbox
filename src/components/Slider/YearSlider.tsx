import PropTypes from 'prop-types';
import * as React from 'react';
import { FlatList, StyleSheet, View } from 'react-native';
import Icon from 'react-native-vector-icons/MaterialIcons';
import colors from '../../styles/colors';
import Chip from './Chip';

const styles = StyleSheet.create({
  container: {
    width: '100%',
    backgroundColor: colors.slider.background,
    alignItems: 'center',
    justifyContent: 'flex-start',
    flexDirection: 'row',
  },
  icon: {
    marginHorizontal: 15,
  },
});

interface YearSliderProps {
  data: ReadonlyArray<number>;
  selected: number;
  onPressSelection: Function;
  refresh?: React.ReactNode;
}

const YearSlider: React.FunctionComponent<YearSliderProps> = ({
  data,
  selected,
  onPressSelection,
  refresh,
}) => (
    <View style={styles.container}>
      <Icon style={styles.icon} size={20} name="access-time" color={colors.white} />
      <FlatList<number>
        horizontal
        showsHorizontalScrollIndicator={false}
        data={data}
        keyExtractor={item => item.toString()}
        extraData={selected}
        renderItem={({ item }) => (
          <Chip
            onPressSelection={onPressSelection}
            selected={selected === item}
            value={item}
          />
        )}
      />
      { refresh && refresh }
    </View>
  );

YearSlider.propTypes = {
  data: PropTypes.array.isRequired,
  selected: PropTypes.number.isRequired,
  onPressSelection: PropTypes.func.isRequired,
  refresh: PropTypes.any,
};

YearSlider.defaultProps = {
  refresh: [],
};

export default YearSlider;
