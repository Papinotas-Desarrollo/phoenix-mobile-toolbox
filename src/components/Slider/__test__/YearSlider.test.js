import * as React from 'react';
import { TouchableOpacity } from 'react-native';
import { fireEvent, render } from 'react-native-testing-library';
import renderer from 'react-test-renderer';
import Chip from '../Chip';
import YearSlider from '../YearSlider';

describe('Snapshot test', () => {
  it('With props snapshot', () => {
    const props = {
      data: [2018, 2019, 2020],
      selected: 2020,
      onPressSelection: jest.fn(),
      refresh: null,
    };
    const tree = renderer.create(<YearSlider {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
describe('Proptype check', () => {
  it('Callback proptypes defined', () => {
    expect(YearSlider.propTypes.data).toBeDefined();
    expect(YearSlider.propTypes.selected).toBeDefined();
    expect(YearSlider.propTypes.onPressSelection).toBeDefined();
    expect(YearSlider.propTypes.refresh).toBeDefined();
  });
});
describe('Functionality', () => {
  it('Three Chip children should be render', () => {
    const props = {
      data: [2018, 2019, 2020],
      selected: 0,
      onPressSelection: jest.fn(),
      refresh: null,
    };
    const { queryAllByType } = render(<YearSlider {...props} />);
    expect(queryAllByType(Chip).length).toBe(3);
  });
  it('Call refetch button', () => {
    const buttonCallback = jest.fn();
    const refreshButton = (
      <TouchableOpacity testID="refreshButton" onPress={buttonCallback} />
    );
    const props = {
      data: [2018, 2019, 2020],
      selected: 0,
      onPressSelection: jest.fn(),
      refresh: refreshButton,
    };
    const { getByTestId } = render(<YearSlider {...props} />);
    fireEvent.press(getByTestId('refreshButton'));
    expect(buttonCallback).toHaveBeenCalled();
  });
});
