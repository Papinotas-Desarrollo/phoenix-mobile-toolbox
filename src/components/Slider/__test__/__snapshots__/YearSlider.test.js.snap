// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test With props snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#75a7e2",
      "flexDirection": "row",
      "justifyContent": "flex-start",
      "width": "100%",
    }
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "#FFFFFF",
          "fontSize": 20,
        },
        Object {
          "marginHorizontal": 15,
        },
        Object {
          "fontFamily": "Material Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    î†’
  </Text>
  <RCTScrollView
    data={
      Array [
        2018,
        2019,
        2020,
      ]
    }
    disableVirtualization={false}
    extraData={2020}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={true}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    showsHorizontalScrollIndicator={false}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "marginHorizontal": 5,
              "opacity": 1,
              "padding": 10,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#A4C5EB",
                  "fontSize": 14,
                },
                null,
              ]
            }
          >
            2018
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "marginHorizontal": 5,
              "opacity": 1,
              "padding": 10,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#A4C5EB",
                  "fontSize": 14,
                },
                null,
              ]
            }
          >
            2019
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "borderBottomColor": "#FF9500",
              "borderBottomWidth": 2,
              "marginHorizontal": 5,
              "opacity": 1,
              "padding": 10,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#A4C5EB",
                  "fontSize": 14,
                },
                Object {
                  "color": "#FFFFFF",
                  "fontWeight": "600",
                },
              ]
            }
          >
            2020
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
